applications/MeshingApplication/tests/Kratos_Execute_Meshing_Test.py

```diff
@@ -19,14 +20,25 @@ def __init__(self, ProjectParameters):
 
         self.Model = {self.ProjectParameters["problem_data"]["model_part_name"].GetString(): self.main_model_part}
 
-        ## Solver construction
-        import python_solvers_wrapper_fluid
-        self.solver = python_solvers_wrapper_fluid.CreateSolver(self.main_model_part, self.ProjectParameters)
+        self.problem_type = self.ProjectParameters["problem_data"]["problem_type"].GetString() 
+    
+        self.solve_problem = self.ProjectParameters["problem_data"]["solve_problem"].GetBool()
+    
+        if self.problem_type  == "fluid":
+            ## Solver construction
+            import python_solvers_wrapper_fluid
+            self.solver = python_solvers_wrapper_fluid.CreateSolver(self.main_model_part, self.ProjectParameters)
+        elif self.problem_type  == "solid":
+            # Construct the solver (main setting methods are located in the solver_module)
+            solver_module = __import__(self.ProjectParameters["solver_settings"]["solver_type"].GetString())
+            self.solver = solver_module.CreateSolver(self.main_model_part, self.ProjectParameters["solver_settings"])
 
         # Add variables (always before importing the model part) (it must be integrated in the ImportModelPart)
         # If we integrate it in the model part we cannot use combined solvers
         self.solver.AddVariables()
         
+        self.main_model_part.AddNodalSolutionStepVariable(NODAL_H) 
```

Comment:
are u checking somewhere within your code that those variables are defined, at least on the first node? if not, do it